hack/install-deps.sh

```diff
@@ -29,25 +29,37 @@ function cmd() {
 # so we can amend our path to look into the local node_modules for the
 # correct binaries.
 repo_root="$( dirname "${BASH_SOURCE}" )/.."
-export PATH="${PATH}:${repo_root}/node_modules/bower/bin:${repo_root}/node_modules/grunt-cli/bin"
+export PATH="${PATH}:${repo_root}/node_modules/grunt-cli/bin"
 
-# Install bower if needed
-if ! which bower > /dev/null 2>&1 ; then
-  cmd "npm install bower"
+# Install yarn if needed
+if ! which yarn > /dev/null 2>&1 ; then
+  cmd "curl -o- -L https://yarnpkg.com/install.sh | bash"
 fi
 
+
+# In case upstream components change things without incrementing versions
+cmd "yarn cache clean"
+cmd "rm -rf node_modules"
+
+# ignore-engines flag is due to a bug in angular-ui-router@1.0.0-beta.1, which
+# is a dependency of origin-web-catalog. This package fails the node engine
+# compatibility test, even when it should not.
+#
+# ignore-scripts flag causes post-install scripts to be skipped. The
+# hawtio-exetnsion-service package has a post-install script which fails
+# because it tries to use bower and gulp which are not dependencies of this
+# project.
+cmd "yarn --ignore-engines --ignore-scripts"
+
+# Work around. --ignore-scripts flag causes phantomjs-prebuilt
+# post install script not be executed which leads to test failure.
```

Comment:
I think we are not using `phantomjs` anymore.  This might be able to be dropped (or spawned as a follow-up issue to avoid getting into weeds right now)

Reply:
If we are not using it, can we remove the dependencies?